packages/api/tests/test-kinesis-consumer.js

```diff
@@ -59,92 +57,125 @@ const disabledRuleParams = Object.assign({}, commonRuleParams, {
+  await s3().putObject({
+    Bucket: t.context.templateBucket,
+    Key: messageTemplateKey,
+    Body: JSON.stringify(t.context.messageTemplate)
+  }).promise();
+
+  sinon.stub(Rule, 'buildPayload').callsFake((item) =>
+    Promise.resolve({
+      template: `s3://${t.context.templateBucket}/${messageTemplateKey}`,
+      provider: item.provider,
+      collection: item.collection,
+      meta: get(item, 'meta', {}),
+      payload: get(item, 'payload', {})
+    })
+  );
+
+  t.context.tableName = randomString();
+  process.env.RulesTable = t.context.tableName;
+  process.env.stackName = randomString();
+  process.env.bucket = randomString();
+  process.env.kinesisConsumer = randomString();
+
+  const model = new Rule(t.context.tableName);
+  await manager.createTable(t.context.tableName, ruleTableParams);
+  await Promise.all([rule1Params, rule2Params, disabledRuleParams]
+    .map((rule) => model.create(rule)));
 });
 
-test.after.always(async () => {
-  await manager.deleteTable(tableName);
+test.afterEach(async (t) => {
+  await Promise.all([
+    recursivelyDeleteS3Bucket(t.context.templateBucket),
+    sqs().deleteQueue({ QueueUrl: t.context.queueUrl }).promise(),
+    manager.deleteTable(t.context.tableName)
+  ]);
+  Rule.buildPayload.restore();
 });
 
 // getKinesisRule tests
-test('it should look up kinesis-type rules which are associated with the collection, but not those that are disabled', t => {
-  return Promise.all([rule1Params, rule2Params, disabledRuleParams].map(x => model.create(x)))
-    .then(() => {
-      return getKinesisRules(JSON.parse(eventData))
-    }).then((result) => {
+// eslint-disable-next-line max-len
+test.serial('it should look up kinesis-type rules which are associated with the collection, but not those that are disabled', (t) => {
+  return getKinesisRules(JSON.parse(eventData))
+    .then((result) => {
       t.is(result.length, 2);
     });
 });
 
 // handler tests
-test('it should create a onetime rule for each associated workflow', async t => {
-  await handler(event, {}, testCallback).then(() => {
-    return model.scan({
-      names: {
-        '#col': 'collection',
-        '#nm': 'name',
-        '#st': 'state',
-        '#rl': 'rule',
-        '#tp': 'type'
-      },
-      filter: '#st = :enabledState AND #col.#nm = :collectionName AND #rl.#tp = :ruleType',
-      values: {
-        ':enabledState': 'ENABLED',
-        ':collectionName': testCollectionName,
-        ':ruleType': 'onetime'
-      }
-    });
-  })
-  .then((results) => {
-    t.is(results.Items.length, 4);
-    const workflowNames = results.Items.map(i => i.workflow).sort();
-    t.deepEqual(workflowNames, [
-      'test-workflow-1',
-      'test-workflow-1',
-      'test-workflow-2',
-      'test-workflow-2'
-    ]);
-    results.Items.forEach(r => t.is(r.rule.type, 'onetime'));
-  });  
+test.serial('it should create a onetime rule for each associated workflow', async (t) => {
```

Comment:
Do we need the serial option here? I thought [we were already running tests in serial](https://github.com/cumulus-nasa/cumulus/blob/master/packages/api/package.json#L7)

Reply:
I put that in when debugging and now have removed it. You're right, nice catch.